integrations/vello_svg/src/lib.rs

```diff
@@ -33,78 +33,107 @@
-/// This will draw a red box over (some) unsupported elements
+/// Append a [`usvg::Tree`] into a Vello [`SceneBuilder`], with default error
+/// handling This will draw a red box over (some) unsupported elements
 ///
 /// Calls [`render_tree_with`] with an error handler implementing the above.
 ///
-/// See the [module level documentation](crate#unsupported-features) for a list of some unsupported svg features
-pub fn render_tree(sb: &mut SceneBuilder, svg: &usvg::Tree) {
-    render_tree_with(sb, svg, default_error_handler).unwrap_or_else(|e| match e {});
+/// See the [module level documentation](crate#unsupported-features) for a list
+/// of some unsupported svg features
+pub fn render_tree(sb: &mut SceneBuilder, svg: &usvg::Tree, transform: Option<Affine>) {
+    render_tree_with(sb, svg, transform, default_error_handler).unwrap_or_else(|e| match e {})
 }
 
 /// Append a [`usvg::Tree`] into a Vello [`SceneBuilder`].
 ///
 /// Calls [`render_tree_with`] with [`default_error_handler`].
 /// This will draw a red box over unsupported element types.
 ///
-/// See the [module level documentation](crate#unsupported-features) for a list of some unsupported svg features
+/// See the [module level documentation](crate#unsupported-features) for a list
+/// of some unsupported svg features
 pub fn render_tree_with<F: FnMut(&mut SceneBuilder, &usvg::Node) -> Result<(), E>, E>(
     sb: &mut SceneBuilder,
     svg: &usvg::Tree,
+    transform2: Option<Affine>,
     mut on_err: F,
 ) -> Result<(), E> {
     for elt in svg.root.descendants() {
         let transform = {
-            let usvg::Transform { a, b, c, d, e, f } = elt.abs_transform();
-            Affine::new([a, b, c, d, e, f])
+            let usvg::Transform {
+                sx,
+                kx,
+                ky,
+                sy,
+                tx,
+                ty,
+            } = elt.abs_transform();
+            let (a, b, c, d, e, f) = (
+                sx as f64, ky as f64, kx as f64, sy as f64, tx as f64, ty as f64,
+            );
+            if let Some(t2) = transform2 {
+                t2 * Affine::new([a, b, c, d, e, f])
+            } else {
+                Affine::new([a, b, c, d, e, f])
+            }
         };
         match &*elt.borrow() {
             usvg::NodeKind::Group(_) => {}
             usvg::NodeKind::Path(path) => {
                 let mut local_path = BezPath::new();
-                // The semantics of SVG paths don't line up with `BezPath`; we must manually track initial points
+                // The semantics of SVG paths don't line up with `BezPath`; we
+                // must manually track initial points
                 let mut just_closed = false;
                 let mut most_recent_initial = (0., 0.);
                 for elt in path.data.segments() {
                     match elt {
-                        usvg::PathSegment::MoveTo { x, y } => {
+                        usvg::tiny_skia_path::PathSegment::MoveTo(p) => {
```

Comment:
This tiny_skia_path identifier is interesting. Do you know the reason it was moved to this path?

Reply:
These changes are detailed here: https://github.com/RazrFalcon/resvg/blob/master/CHANGELOG.md#0340---2023-05-27

Unfortunately the move to TinySkia and f32s in general means a ton of f64 casts.